### PR TITLE
(DOCSP-28906) Add core logging docs for Node.js SDK

### PR DESCRIPTION
**Pull request by krollins-mdb**
## Pull Request Info

DO NOT MERGE until v12 GA ships. After release:

- [x] Add links to generated API documentation for new APIs

### Jira

- https://jira.mongodb.org/browse/DOCSP-28906

### Staged Changes

- [Logging](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-28906-nodejs/sdk/node/logging/). New page that outlines the new feature.
- [Set the Client Log Level](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-28906-nodejs/sdk/node/sync/log-level/). Add warning that points readers to the new core logger.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Review by dacharyc — APPROVED** [review]
Pending the API links, LGTM!

**Review by kneth — APPROVED** [review]
I like the way you test it. 

Question: is it worth mentioning that `console.log()` on node.js is synchronous and should be avoided? See for example https://facsiaginsa.com/nodejs/how-logging-affect-performance-fastify for implications on performance.

File: source/sdk/node/logging.txt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+the log level, pass :js-sdk:`Realm.setLogLevel() <TBD>` to a valid
+:js-sdk:`RealmLogLevel <TBD>` level value.
+
+.. literalinclude:: /examples/generated/node/v12/logger.test.snippet.set-log-level.ts
+   :language: typescript
+
+You can set different log levels to increase or decrease verbosity at different
+points in your code. This behavior differs from the deprecated 
+:ref:`sync client log level <node-client-log-level>`, which had to be set
+before opening a synced realm and could not be changed.
```
Comment (takameyer) [inline source/sdk/node/logging.txt:41]: I don't think it's worth adding the verbage here to compare to the deprecated logger.  Perhaps a section on "migrating from sync logger" would be more appropriate.  Just a thought

**Review by takameyer — APPROVED** [review]
Looks great!  Just have one consideration.

File: source/sdk/node/logging.txt (hunk 1 above)
Comment (krollins-mdb) [reply source/sdk/node/logging.txt:41]: We're using similar language in the other SDK docs. We'll remove it after the new logger has been around for a while.

Comment (takameyer) [reply source/sdk/node/logging.txt:41]: Sounds good!

**Pull request merged**